### PR TITLE
Feature: model preview selection

### PR DESCRIPTION
**Pull request by NicolasRichel**
https://platform-dev-model-preview-selection.bimdata.io

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have tested my code.
- [x] I want to run the tests for the commits of this PR

### CODE REVIEW

File: src/components/specific/models/models-card/main-model-selector/MainModelSelector.vue [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  () => {
+    selectedModel.value =
+      props.models.find(m => m.id === props.project.main_model_id) ??
+      props.models[0];
+  },
+  { immediate: true }
+);
+
+const cancel = () => {
+  props.onCancel?.();
```
Comment (Amoki) [inline src/components/specific/models/models-card/main-model-selector/MainModelSelector.vue]: Are there cases where onCancel in not defined?

I prefer non-nullable props than `?.` everywhere when possible

File: src/components/specific/models/models-card/models-card-model-preview/ModelsCardModelPreview.vue [hunk 2]
```diff
@@ -1,7 +1,36 @@
...
+          class="file-actions-cell__btn"
+          ripple
+          rounded
+          icon
+          @click="isOpenMenu = !isOpenMenu"
+        >
+          <BIMDataIconEllipsis size="l" />
+        </BIMDataButton>
+
+        <BIMDataMenu
```
Comment (Amoki) [inline src/components/specific/models/models-card/models-card-model-preview/ModelsCardModelPreview.vue:20]: show only for admins?

**Review by Amoki — CHANGES_REQUESTED** [review]

**Review by Kurtil — APPROVED** [review]

File: src/components/specific/models/models-card/main-model-selector/MainModelSelector.vue (hunk 1 above)
Comment (Kurtil) [reply src/components/specific/models/models-card/main-model-selector/MainModelSelector.vue]: or a default props to `() => {}`

Comment (Amoki) [reply src/components/specific/models/models-card/main-model-selector/MainModelSelector.vue]: I think I prefer `?.` than `() => {}` defined somewhere else.

But I don't why we would need nullable fields on a component used in one place

**Review by Amoki — APPROVED** [review]

**Review by Amoki — APPROVED** [review]

**Review by GabBimdata — APPROVED** [review]

**Pull request merged**